firstvoices/backend/models/characters.py

```diff
@@ -80,6 +82,14 @@ def validate_title_uniqueness(self):
                 % self.title
             )
 
+    def validate_character_limit(self):
+        """
+        Validates that the site has not already reached the max character limit.
+        """
+        limit = CustomSorter.max_alphabet_length
+        if len(Character.objects.filter(site_id=self.site_id)) >= limit:
```

Comment:
Minor optimization: instead of doing len() you can add .count() to the queryset, if you don't need to retrieve all the actual rows of data.